Development/OpenPassDevelopmentApp/DeviceAuthorizationViewModel.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  DeviceAuthorizationViewModel.swift
+//
+// MIT License
+//
+// Copyright (c) 2024 The Trade Desk (https://www.thetradedesk.com/)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+#if os(tvOS)
+
+import CoreImage.CIFilterBuiltins
+import Foundation
+import OpenPass
+import SwiftUI
+
+extension DeviceAuthorizationViewModel {
+
+    /// A interface defining the flow of state communicated by the `DeviceAuthorizationFlow`
+    public enum State: Sendable {
+
+        /// The client has been initialized but a ``DeviceCode`` has not been requested or received
+        case initial
+
+        case deviceCodeAvailable(DeviceCode)
+
+        /// The previous ``DeviceCode`` has now expired, and the consumer is required to re-start the flow via
+        /// ``DeviceAuthorizationFlow.fetchDeviceCode()``.
+        case deviceCodeExpired
+
+        /// An unexpected error has occurred.
+        case error(Error)
+
+        /// The flow is complete and the associated ``OpenPassManager`` has obtained the set of ``OpenPassTokens``.
+        case complete(OpenPassTokens)
+    }
+}
+
+@MainActor
+final class DeviceAuthorizationViewModel: ObservableObject {
+
+    @Published private(set) var deviceCode: DeviceCode?
+
+    @Published private(set) var verificationUriCompleteImage: UIImage?
+
+    @Published private(set) var error: Error?
+
+    @Published private(set) var state: DeviceAuthorizationViewModel.State = .initial {
+        didSet {
+            switch state {
+            case .initial:
+                break
+            case .deviceCodeAvailable(let deviceCode):
+                self.deviceCode = deviceCode
+                if let verificationUriComplete = deviceCode.verificationUriComplete {
+                    verificationUriCompleteImage = qrCodeImage(url: verificationUriComplete)
```

Comment:
Nit: In other instances, you used `self.` but not in this one. Is there a reason for that?